lib/dotcom/trip_plan/itinerary_groups.ex

```diff
@@ -8,48 +8,34 @@ defmodule Dotcom.TripPlan.ItineraryGroups do
 
   import DotcomWeb.TripPlanView, only: [get_one_way_total_by_type: 2]
 
-  alias Dotcom.TripPlan.{Itinerary, Leg, PersonalDetail, TransitDetail}
+  alias Dotcom.TripPlan.{Itinerary, ItineraryGroup, Leg, PersonalDetail, TransitDetail}
   alias OpenTripPlannerClient.ItineraryTag
 
   @short_walk_threshold_minutes 5
-  @max_per_group 5
-
-  @type summarized_leg :: %{
-          routes: [Routes.Route.t()],
-          walk_minutes: non_neg_integer()
-        }
-  @type summary :: %{
-          accessible?: boolean() | nil,
-          duration: non_neg_integer(),
-          first_start: DateTime.t(),
-          first_stop: DateTime.t(),
-          next_starts: [DateTime.t()],
-          summarized_legs: [summarized_leg()],
-          tag: String.t(),
-          total_cost: non_neg_integer(),
-          walk_distance: float()
-        }
-
-  @spec from_itineraries([Itinerary.t()]) :: [
-          %{itineraries: [Itinerary.t()], summary: summary()}
-        ]
-  def from_itineraries(itineraries) do
+  @max_per_group 4
+
+  @doc """
+  From a large list of itineraries, collect them into 5 groups of at most
+  #{@max_per_group} itineraries each, sorting the groups in favor of tagged
+  groups first
+  """
+  @spec from_itineraries([Itinerary.t()], Keyword.t()) :: [ItineraryGroup.t()]
+  def from_itineraries(itineraries, opts \\ []) do
     itineraries
     |> Enum.group_by(&unique_legs_to_hash/1)
     |> Enum.map(&drop_hash/1)
     |> Enum.reject(&Enum.empty?/1)
-    |> Enum.take(5)
-    |> Enum.map(&limit_itinerary_count/1)
-    |> Enum.map(&to_summarized_group/1)
+    |> Enum.map(&limit_itinerary_count(&1, opts))
+    |> Enum.map(&to_group(&1, opts))
     |> Enum.sort_by(fn
-      %{itineraries: [%{tag: tag} | _] = _} ->
+      %ItineraryGroup{summary: %{tag: tag}} ->
         Enum.find_index(ItineraryTag.tag_priority_order(), &(&1 == tag))
-
-      _ ->
-        -1
     end)
+    |> Enum.take(5)
   end
 
+  def max_per_group, do: @max_per_group
```

Comment:
Nit (non-blocking): This is bugging me a bit - the only reason it exists is to expose this value to the tests, but I feel like the tests can afford to be explicit about the fact that the itinerary-count-limit is 4.